src/plugins_3rdparty/primer3/src/Primer3Task.cpp

```diff
@@ -25,114 +25,216 @@
 #include <U2Core/Counter.h>
 #include <U2Core/CreateAnnotationTask.h>
 #include <U2Core/DNASequenceObject.h>
+#include <U2Core/L10n.h>
 #include <U2Core/MultiTask.h>
 #include <U2Core/SequenceWalkerTask.h>
 #include <U2Core/U1AnnotationUtils.h>
 
+#include "Primer3Plugin.h"
 #include "Primer3Task.h"
-#include "primer3_core/boulder_input.h"
-#include "primer3_core/primer3_main.h"
+
+#include "primer3_core/primer3_boulder_main.h"
+#include "primer3_core/print_boulder.h"
+#include "primer3_core/p3_seq_lib.h"
+#include "primer3_core/libprimer3.h"
 
 namespace U2 {
 
-// Primer
+// PrimerSingle
 
-Primer::Primer()
+PrimerSingle::PrimerSingle(oligo_type type)
     : start(0),
       length(0),
       meltingTemperature(0),
+      bound(0),
```

Comment:
initialize primitive fields in the header, so you do not need to repeat and can't forget

Reply:
OK